examples/hello_world/CMakeLists.txt

```diff
@@ -10,26 +10,29 @@ find_package(Threads REQUIRED)
 
 set(VSOMEIP_NAME "vsomeip3")
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+# create_target("executable")
+function(create_target executable)
+    add_library(vsomeip_hello_world_${executable} INTERFACE)
+    target_sources(vsomeip_hello_world_${executable} INTERFACE
+        "${CMAKE_CURRENT_SOURCE_DIR}/hello_world_${executable}.hpp"
+    )
+    target_compile_features(vsomeip_hello_world_${executable} INTERFACE cxx_std_17)
+
+    target_include_directories(vsomeip_hello_world_${executable} INTERFACE
+        "${CMAKE_CURRENT_SOURCE_DIR}"
+    )
+endfunction()
+
+# link_target("executable")
+function(link_target executable)
+    add_executable (hello_world_${executable} hello_world_${executable}_main.cpp)
+    target_link_libraries(hello_world_${executable} vsomeip_hello_world_${executable} ${VSOMEIP_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
`${CMAKE_THREAD_LIBS_INIT}` should be replaced with `Theads::Threads` from `find_package(Threads)`